lms/djangoapps/email_marketing/signals.py

```diff
@@ -263,7 +264,23 @@ def _log_sailthru_api_call_time(time_before_call):
              delta_sailthru_api_call_time.microseconds / 1000)
 
 
+def _backoff_handler(details):
+    """
+    Simple logging handler for when timeout backoff occurs.
+    """
+    log.info('Trying again in {wait:0.1f} seconds after {tries} tries calling {target}'.format(**details))
+
+
+def _not_a_timeout(e):
+    return 'timed out' not in e.message
+
+
 @receiver(USER_RETIRE_MAILINGS)
+@backoff.on_exception(backoff.expo,
+                      Exception,
+                      max_time=600,  # Only 10 minutes of trying
```

Comment:
If we have a big downstream downtime this means all retirement jobs will hang for 10 mins+, correct? Is that ok, or will it tie up Jenkins workers that also do other thing?

Reply:
It will tie up a jenkins-worker instance for 10 minutes, and if we allow 4 concurrent retirements, 4 jenkins-worker instances will be tied up.  I'm not too worried about that, considering there looks to be 40-50 active jenkins-worker cranking out PR tests at any given time.